### ISSUE TITLE
fish: No matches for wildcard '--include=*.xml'  when use grep

### ISSUE THREAD

**Issue by ghost** [fish-shell/fish-shell#683]
Hi,when I use command grep like
`grep -rn 'pathis' --include=*.xml ./`
I got this error 
`fish: No matches for wildcard '--include=*.xml'`

this command in bash is OK
I use xubuntu12.04, xfce Terminal 0.4.8

**Comment by bucaran** [comment]
I was trying to do something like this:

```fish
# Traverse $path AND $path subdirs
for dir in path /Users/me/path/**/
   echo "Don't even bother printing $dir if $dir has no subdirs"
end
```

Apparently due to the wildcar returning not matches `/**/` the for loop was __not even__ running for `$path`, the parent directory. Escaping the wildcars as suggested didn't fix anything, but the following did:

```fish
  set -l subdirs $path/**/
  # Traverse $path and prepend only directories with matches.
  for dir in $path $subdirs
     ....
```

Incidentally I found the above rather more logical than what I was doing earlier. Fish FTW.

**Comment by krader1961** [comment]
This has been addressed by issue #1482 and issue #2719 so I'm closing this issue.
